components/infobox/wikis/overwatch/infobox_person_player_custom.lua

```diff
@@ -0,0 +1,217 @@
+	['analyst'] = {category = 'Analysts', variable = 'Analyst', isplayer = false},
+	['observer'] = {category = 'Observers', variable = 'Observer', isplayer = false},
+	['host'] = {category = 'Hosts', variable = 'Host', isplayer = false},
+	['journalist'] = {category = 'Journalists', variable = 'Journalist', isplayer = false},
+	['expert'] = {category = 'Experts', variable = 'Expert', isplayer = false},
+	['coach'] = {category = 'Coaches', variable = 'Coach', isplayer = false},
+	['caster'] = {category = 'Casters', variable = 'Caster', isplayer = false},
+	['talent'] = {category = 'Talents', variable = 'Talent', isplayer = false},
+	['manager'] = {category = 'Managers', variable = 'Manager', isplayer = false},
+	['producer'] = {category = 'Producers', variable = 'Producer', isplayer = false},
+	['admin'] = {category = 'Admins', variable = 'Admin', isplayer = false},
+}
+_ROLES['assistant coach'] = _ROLES.coach
+
+local _SIZE_HERO = '25x25px'
+
+local CustomPlayer = Class.new()
+
+local CustomInjector = Class.new(Injector)
+
+local _pagename = mw.title.getCurrentTitle().prefixedText
+local _args
+local _player
+
+function CustomPlayer.run(frame)
+	local player = Player(frame)
+
+	player.args.history = tostring(TeamHistoryAuto._results{addlpdbdata='true'})
+
+	player.adjustLPDB = CustomPlayer.adjustLPDB
+	player.createWidgetInjector = CustomPlayer.createWidgetInjector
+	player.defineCustomPageVariables = CustomPlayer.defineCustomPageVariables
+
+	_args = player.args
+	_player = player
+
+	return player:createInfobox(frame)
+end
+
+function CustomPlayer:createWidgetInjector()
+	return CustomInjector()
+end
+
+function CustomInjector:parse(id, widgets)
+	if id == 'status' then
+		return {
+			Cell{name = 'Status', content = CustomPlayer._getStatusContents()},
+			Cell{name = 'Years Active (Player)', content = {_args.years_active}},
+			Cell{name = 'Years Active (Org)', content = {_args.years_active_manage}},
+			Cell{name = 'Years Active (Coach)', content = {_args.years_active_coach}},
+			Cell{name = 'Years Active (Talent)', content = {_args.years_active_talent}},
+			Cell{name = 'Time Banned', content = {_args.time_banned}},
+		}
+	elseif id == 'role' then
+		return {
+			Cell{name = 'Role', content = {
+				CustomPlayer._createRole('role', _args.role),
+				CustomPlayer._createRole('role2', _args.role2)
+			}},
+		}
+	elseif id == 'history' then
+		table.insert(widgets, Cell{
+			name = 'Retired',
+			content = {_args.retired}
+		})
+	end
+	return widgets
+end
+
+function CustomInjector:addCustomCells(widgets)
+	-- Signature Heroes
+	local heroIcons = Array.map(Player:getAllArgsForBase(_args, 'hero'),
+		function(hero, _)
+			local standardizedHero = HeroNames[hero:lower()]
+			if not standardizedHero then
+				-- we have an invalid hero entry
+				-- add warning (including tracking category)
+				table.insert(
+					_player.warnings,
+					'Invalid hero input "' .. hero .. '"[[Category:Pages with invalid hero input]]'
+				)
+			end
```

Comment:
Looks to me that getImage in HeroIcon already does this

Reply:
So just remove this and allow HeroIcon to hold the error?